x-pack/plugins/search_inference_endpoints/public/components/all_inference_endpoints/use_all_inference_endpoints_state.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useState } from 'react';
+
+import type { QueryParams, AlInferenceEndpointsTableState } from './types';
+
+import { DEFAULT_INFERENCE_ENDPOINTS_TABLE_STATE } from './constants';
+
+interface UseAllInferenceEndpointsStateReturn {
+  queryParams: QueryParams;
+  setQueryParams: (queryParam: Partial<QueryParams>) => void;
+}
+
+export function useAllInferenceEndpointsState(): UseAllInferenceEndpointsStateReturn {
```

Comment:
we should define this file in the `hooks/` directory

Reply:
Updated